xarray/tests/test_dataarray.py

```diff
@@ -1874,6 +1874,13 @@ def test_array_interface(self):
         bar = Variable(["x", "y"], np.zeros((10, 20)))
         assert_equal(self.dv, np.maximum(self.dv, bar))
 
+    def test_astype_attrs(self):
+        mda1 = self.mda.copy()
+        mda1.attrs["foo"] = "bar"
+        mda2 = mda1.astype(bool)
```

Comment:
Do we already have tests for `.astype` working correctly? If not, could we add a couple of simple cases (no need to do too much tbc)